Platform/AlderlakeBoardPkg/Library/ShellExtensionLib/CmdGpio.c

```diff
@@ -35,6 +100,98 @@ CONST SHELL_COMMAND mShellCommandGpio = {
   &ShellCommandGpioFunc
 };
 
+/**
+  Calculate GPIO group value from group array index
+
+  @param[in]  GrpArrayIndex GPIO group array index
+  @param[out] GpioGroup     GPIO group
+**/
+EFI_STATUS
+EFIAPI
+GpioGetGroup(
+  IN UINT32 GrpArrayIndex,
+  OUT UINT32 *GpioGroup)
+{
+  if (IsPchS()) {
+    *GpioGroup = GpioGrpAdlsTable[GrpArrayIndex];
+  }
+  else if (IsPchP() || IsPchN()) {
+    *GpioGroup = GpioGrpAdlnpTable[GrpArrayIndex];
+  }
+
+  if (*GpioGroup == 0) {
```

Comment:
*GpioGroup isn't guaranteed to be initialized to zero in this or the calling function. Can you please add a different check for the failing case?

Reply:
Will assign *GpioGroup value to zero in this function. After that *GpioGroup is getting assigned to GPIO group value from hardcoded global array corresponding to PCH and GrpArrayIndex. If GrpArrayIndex mapped to value zero in the global array, then that group is not present in ADL platform.